pipelines/controller/src/main/java/com/google/fhir/analytics/FlinkConfiguration.java

```diff
@@ -41,10 +41,24 @@ public class FlinkConfiguration {
   private static final Logger logger = LoggerFactory.getLogger(FlinkConfiguration.class.getName());
   static final String TEMP_FLINK_CONF_DIR = "tmp-flink-conf";
 
-  // This value is arrived based on the maximum number of reshuffle/partition operations that can
-  // occur in the pipelines. This has to be updated if any modifications are made to the pipelines
-  // that impact the reshuffle/partition operations.
+  /**
+   * This value is arrived based on the maximum number of reshuffle/partition operations that can
+   * occur in the pipelines. This has to be updated if any modifications are made to the pipelines
+   * that impact the reshuffle/partition operations.
+   *
+   * <p>Currently, for incremental run a maximum of 7 reshuffle/partition operations take place.
+   *
+   * <ul>
+   *   <li>2 reshuffle operations for matching and reading from Parquet files
+   *   <li>1 reshuffle operation for GroupByKey
+   *   <li>4 reshuffle operations for writing back to new parquet files (some kind of map-reduce
```

Comment:
Can you please file a bug for this such that we investigate it in future? We should not really need 4 reshuffles there (it is ParquetIO that is doing this and maybe we can fix it or configure it to avoid this).

Reply:
Have created this [issue](https://github.com/google/fhir-data-pipes/issues/891) for the same.